health-services/project-factory/src/server/kafka/Producer.ts

```
@@ -91,8 +91,13 @@ const sendWithReconnect = (payloads: any[]): Promise<void> => {
 };
 
 
-async function produceModifiedMessages(modifiedMessages: any[], topic: any) {
+async function produceModifiedMessages(modifiedMessages: any[], topic: any, tenantId: string = config.app.defaultTenantId): Promise<void> {
     try {
+        if(config.isEnvironmentCentralInstance) {
+            // If tenantId has no ".", default to tenantId itself
+            const firstTenantPartAfterSplit = tenantId.includes(".") ? tenantId.split(".")[0] : tenantId;
+            topic = `${firstTenantPartAfterSplit}-${topic}`;
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)